### PR TITLE
convert log to use fetch streaming

### PR DESCRIPTION
**Pull request by davidgamero**
remove another remaining usage of the request library
ref #754

### CODE REVIEW

File: src/log.ts [hunk 1]
```diff
@@ -90,38 +113,33 @@ export class Log {
...
-                        reject(new HttpError(response, body, response.statusCode));
+        const requestOptions: RequestOptions = {};
+
+        const requestURL = new URL(cluster.server + path);
+
+        var searchParams = requestURL.searchParams;
+        AddOptionsToSearchParams(options, searchParams);
+
+        await this.config.applytoHTTPSOptions(requestOptions);
+        console.log(requestURL.toString())
```
Comment (brendandburns) [inline src/log.ts:124]: Remove this stray debug code.

File: src/log.ts [hunk 2]
```diff
@@ -90,38 +113,33 @@ export class Log {
...
+        const req = await fetch(requestURL.toString(), requestOptions)
+            .then(response => {
+                const status = response.status;
+                if (status == 200) {
```
Comment (brendandburns) [inline src/log.ts:128]: probably not strictly necessary, but triple equals here?

**Comment by brendandburns** [comment]
Couple of comments, is there a unit test that needs updating also?

File: src/log.ts (hunk 2 above)
Comment (davidgamero) [reply src/log.ts]: got it

**Comment by davidgamero** [comment]
> Couple of comments, is there a unit test that needs updating also?

i don't believe there was a log_test, but i added one for param parsing

**Pull request merged**